tests/Feature/Conversations/Messages/FilterTest.php

```diff
@@ -0,0 +1,123 @@
+
+        $response = $this->makeRequest($user, $oldMessage->conversation, [
+            'filter' => [
+                'created_after' => now()->toDateTimeString(),
+            ],
+        ]);
+
+        $this->assertResponse($response, $newMessage, $oldMessage);
+    }
+
+    /** @test */
+    public function user_can_index_messages_after_updating_date(): void
+    {
+        [$user, $oldMessage, $newMessage] = $this->prepare('updated_at');
+
+        $response = $this->makeRequest($user, $oldMessage->conversation, [
+            'filter' => [
+                'updated_after' => now()->toDateTimeString(),
+            ],
+        ]);
+
+        $this->assertResponse($response, $newMessage, $oldMessage);
+    }
+
+    /**
+     * Prepares for tests.
+     *
+     * @param  string  $field
+     * @return array
+     */
+    private function prepare(string $field = 'created_at'): array
+    {
+        $user = User::factory()->create();
+
+        $conversation = Conversation::factory()->create();
+
+        // Add user as participant.
+        $conversation->users()->attach($user);
+
+        $oldMessage = Message::factory()->create([
+            'conversation_id' => $conversation->id,
+            $field => now()->subWeek(),
+        ]);
+
+        $newMessage = Message::factory()->create([
+            'conversation_id' => $conversation->id,
+            $field => now()->addWeek(),
+        ]);
+
+        return [$user, $oldMessage, $newMessage, $conversation];
+    }
+
+    /**
+     * Makes a request.
+     *
+     * @param  \OwowAgency\Gossip\Tests\Support\Models\User  $user
+     * @param  \OwowAgency\Gossip\Models\Conversation  $conversation
+     * @param  array  $params
+     * @return \Illuminate\Testing\TestResponse
+     */
+    private function makeRequest(
+        User $user,
+        Conversation $conversation,
+        array $params = []
+    ): TestResponse {
+        $url = "/conversations/{$conversation->id}/messages?" . http_build_query($params);
+
+        return $this->actingAs($user)
+            ->json('GET', $url);
+    }
+
+    /**
+     * Asserts a response.
+     *
+     * @param  \Illuminate\Testing\TestResponse  $response
+     * @param  \OwowAgency\Gossip\Models\Message  $assert
+     * @param  \OwowAgency\Gossip\Models\Message  $missing
+     * @param  int  $status
+     * @return void
+     */
+    private function assertResponse(
+        TestResponse $response,
+        Message $assert,
+        Message $missing,
+        int $status = 200
+    ): void {
+        $response->assertStatus($status);
+
+        if ($status !== 200) {
+            return;
+        }
+
+        $this->assertJsonStructureSnapshot($response);
+
+        $response->assertJson([
+            'data' => [
+                ['id' => $assert->id],
+            ]
+        ]);
+
+        $response->assertJsonMissing([
+            'data' => [
+                ['id' => $missing->id],
+            ]
+        ]);
```

Comment:
No need for this. `assertJsonStructureSnapshot` checks id properties to be exact value rather than only type.